domain/model/service/service.go

```diff
@@ -542,3 +569,56 @@ func (s *Service) UpdateCredential(
 
 	return s.st.UpdateCredential(ctx, uuid, key)
 }
+
+// Watch returns a watcher that monitors changes to models.
+// The watcher listens for create and update events for all models but processes
+// only the activated ones. It maps change events using a filtering function that:
+//  1. Extracts the model UUID from each create/update event.
+//  2. Retrieve and filter UUIDs of activated models from all the extracted UUIDs.
+//  3. Includes the event in the watcher stream associated with each filtered UUID.
+//
+// Potential concerns:
+//   - If retrieving the UUIDs of all activated models from the input list of UUIDs fails,
+//     these events would be skipped, which may lead to missed updates if errors occur frequently.
+//   - This approach may also include change events of already activated models, eg. updating the model name.
+//
+// Returns:
+//   - watcher.StringsWatcher: A watcher that streams events related to activated models.
+//   - error: An error if the watcher cannot be created.
+func (s *Service) Watch(ctx context.Context) (watcher.StringsWatcher, error) {
```

Comment:
I think a lot of the comments above exist because we haven't named the function well. I would consider renaming the function to `WatchActivatedModels`.

Then the description of the func could become something like the following:

```
WatchActivatedModels returns a watcher that reports all activated models in the controller. The watcher will also output events for when an activated model receives an update. Deletion of activated models won't be reported.
```